docker/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Dockerfile to build the matrixdotorg/sygnal docker images.
+#
+# To build the image, run `docker build` command from the root of the
+# sygnal repository:
+#
+#    docker build -f docker/Dockerfile .
+#
+
+###
+### Stage 0: builder
+###
+FROM python:3.7-slim as builder
+
+# build things which have slow build steps, before we copy sygnal, so that
+# the layer can be cached.
+#
+# (we really just care about caching a wheel here, as the "pip install" below
+# will install them again.)
+
+RUN pip install --prefix="/install" --no-warn-script-location cryptography
+
+# now install sygnal and all of the python deps to /install.
+
+COPY sygnal /sygnal/sygnal/
+COPY MANIFEST.in README.rst VERSION setup.py /sygnal/
```

Comment:
I'd be somewhat tempted to copy the entire working directory and `pip install` that so we don't have to worry about forgetting to add any future extra files here. Yes, it'll copy any silly stuff in your local copy, but that won't end up in the final docker image.

Reply:
It's a wise decision, the `.git` folder is now needed to build newer releases